test/fastify-plugins/maps.js

```diff
@@ -223,6 +225,83 @@ test('/style.json resolves style.json of fallback map when custom and online are
   )
 })
 
+test('custom map info endpoint not available when custom map path not provided', async (t) => {
+  const server = setup(t)
+
+  server.register(MapServerPlugin, {
+    defaultOnlineStyleUrl: DEFAULT_ONLINE_STYLE_URL,
+    fallbackMapPath: DEFAULT_FALLBACK_MAP_FILE_PATH,
+  })
+
+  const address = await server.listen()
+
+  mockAgent.enableNetConnect(new URL(address).host)
+
+  const response = await server.inject({
+    method: 'GET',
+    url: `/${CUSTOM_MAP_PREFIX}/info`,
+  })
+
+  assert.equal(response.statusCode, 404)
+})
+
+test('custom map info endpoint returns expected error when custom map is invalid', async (t) => {
+  const server = setup(t)
+
+  const nonExistentFile =
+    path.parse(SAMPLE_SMP_FIXTURE_PATH).dir + '/does/not/exist.smp'
+
+  server.register(MapServerPlugin, {
+    customMapPath: nonExistentFile,
+    defaultOnlineStyleUrl: DEFAULT_ONLINE_STYLE_URL,
+    fallbackMapPath: DEFAULT_FALLBACK_MAP_FILE_PATH,
+  })
+
+  const address = await server.listen()
+
+  mockAgent.enableNetConnect(new URL(address).host)
+
+  const response = await server.inject({
+    method: 'GET',
+    url: `/${CUSTOM_MAP_PREFIX}/info`,
+  })
+
+  assert.equal(response.statusCode, 500, 'returns server error status code')
```

Comment:
helpful maybe to distinguish "custom map file does not exist" (404) from "custom map file is invalid" (500), since at some point we will probably want different UX for each type, and before then we might want to report actual 500 errors to Sentry, but not 404.